test/jdk/java/net/httpclient/HttpClientClose.java

```diff
@@ -0,0 +1,345 @@
+                    Thread.sleep(sleep);
+                }
+                bodyCF.get();
+            }
+        } catch (Throwable throwable) {
+            failed = throwable;
+        } finally {
+            failed = cleanup(readerService, failed);
+        }
+        if (failed instanceof Exception ex) throw ex;
+        if (failed instanceof Error e) throw e;
+        assertTrue(toCheck.isTerminated());
+    }
+
+    // -- Infrastructure
+
+    @BeforeTest
+    public void setup() throws Exception {
+        out.println("\n**** Setup ****\n");
+        sslContext = new SimpleSSLContext().get();
+        if (sslContext == null)
+            throw new AssertionError("Unexpected null sslContext");
+
+        httpTestServer = HttpTestServer.create(HTTP_1_1);
+        httpTestServer.addHandler(new ServerRequestHandler(), "/http1/exec/");
+        httpURI = "http://" + httpTestServer.serverAuthority() + "/http1/exec/retry";
+        httpsTestServer = HttpTestServer.create(HTTP_1_1, sslContext);
+        httpsTestServer.addHandler(new ServerRequestHandler(),"/https1/exec/");
+        httpsURI = "https://" + httpsTestServer.serverAuthority() + "/https1/exec/retry";
+
+        http2TestServer = HttpTestServer.create(HTTP_2);
+        http2TestServer.addHandler(new ServerRequestHandler(), "/http2/exec/");
+        http2URI = "http://" + http2TestServer.serverAuthority() + "/http2/exec/retry";
+        https2TestServer = HttpTestServer.create(HTTP_2, sslContext);
+        https2TestServer.addHandler(new ServerRequestHandler(), "/https2/exec/");
+        https2URI = "https://" + https2TestServer.serverAuthority() + "/https2/exec/retry";
+
+        httpTestServer.start();
+        httpsTestServer.start();
+        http2TestServer.start();
+        https2TestServer.start();
+    }
+
+    @AfterTest
+    public void teardown() throws Exception {
+        Thread.sleep(100);
+        AssertionError fail = TRACKER.checkShutdown(5000);
+        try {
+            httpTestServer.stop();
+            httpsTestServer.stop();
+            http2TestServer.stop();
+            https2TestServer.stop();
+        } finally {
+            if (fail != null) throw fail;
+        }
+    }
+
+    static class ServerRequestHandler implements HttpTestHandler {
+        ConcurrentHashMap<String,String> closedRequests = new ConcurrentHashMap<>();
+
+        @java.lang.Override
+        public void handle(HttpTestExchange t) throws IOException {
+            out.println("ServerRequestHandler for: " + t.getRequestURI());
+
+            List<String> uuids = t.getRequestHeaders().get("X-uuid");
+            if (uuids == null || uuids.size() != 1) {
+                readAllRequestData(t);
+                try (OutputStream os = t.getResponseBody()) {
+                    String msg = "Incorrect uuid header values:[" + uuids + "]";
+                    (new RuntimeException(msg)).printStackTrace();
+                    t.sendResponseHeaders(500, -1);
```

Comment:
Same comment about the `-1` value.

Reply:
Right - same answer: there's no harm in using chunked mode.